### PR TITLE
feat(queries-preview): add wiring and actions

### PR DESCRIPTION
**Pull request by mnavarroespinosa**
[EX-6844](https://searchbroker.atlassian.net/browse/EX-6844)

## Motivation and context
Added wiring and actions to the `queries-preview` module.

- The module's state now stores a dictionary of `QueryPreviewItems`, where each item contains, mainly, the results of a specific query and the request used for the search.

- `fetchQueryPreview` and `fetchAndSaveQueryPreview` actions have been added in order to perform the request and save the results, and the request, used for the search.

#### Events
- Whenever the `QueryPreviewRequestChange` is emitted, a search will be performed with the passed request.
> The component described by [EX-6845](https://searchbroker.atlassian.net/browse/EX-6845) will be in charge of emitting this event if the `extraParams` change.

- `QueryPreviewRemoved` event will remove the entry of a specific query from the `queriesPreview` dictionary.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

### CODE REVIEW

File: packages/x-components/src/x-modules/queries-preview/events.types.ts [hunk 1]
```diff
@@ -1,7 +1,20 @@
...
 /**
  * Dictionary of the events of QueriesPreview XModule, where each key is the event name, and the
  * value is the event payload type or `void` if it has no payload.
  *
  * @public
  */
-export interface QueriesPreviewXEvents {}
+export interface QueriesPreviewXEvents {
+  /**
+   * Query preview request have been changed.
```
Comment (tiborux) [inline packages/x-components/src/x-modules/queries-preview/events.types.ts:11]:
```suggestion
   * Any property of the queries preview request has changed.
```

To match other modules

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  describe('fetchAndSaveQueryPreview', () => {
+    it('should request and store query preview results in the state', async () => {
+      const query = 'tshirt';
+      const request = getQueryPreviewRequest(query);
+
+      const promise = store.dispatch('fetchAndSaveQueryPreview', request);
+      await promise;
+
+      const expectedResults = {
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:71]: For complex types that are no inferred like this one, it is always better to have them explicitly typed to ease future refactors.

```suggestion
      const expectedResults:  QueryPreviewItem = {
```

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    });
+  });
+
+  describe('fetchAndSaveQueryPreview', () => {
+    it('should request and store query preview results in the state', async () => {
+      const query = 'tshirt';
+      const request = getQueryPreviewRequest(query);
+
+      const promise = store.dispatch('fetchAndSaveQueryPreview', request);
+      await promise;
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:69]: Why store the promise and await it in another line?

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  localVue.config.productionTip = false; // Silent production console messages.
+  localVue.use(Vuex);
+
+  const store: SafeStore<
+    QueriesPreviewState,
+    QueriesPreviewGetters,
+    QueriesPreviewMutations,
+    QueriesPreviewActions
+  > = new Store(queriesPreviewXStoreModule as any);
+  installNewXPlugin({ adapter, store }, localVue);
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:39]: You are creating a global store, and you never reset its state. Therefore each test case result is dependant from the previous test. You can either do something similar like we do with component render factories, or reset the store state before every test.

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+      expect(results).toEqual(mockedSearchResponse);
+    });
+
+    it('should return `null` if the query is empty', async () => {
+      const request = getQueryPreviewRequest('');
+      expect(await store.dispatch('fetchQueryPreview', request)).toBeNull();
+    });
+  });
+
+  describe('fetchAndSaveQueryPreview', () => {
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:63]: Can we have some tests asserting the different status values?

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/queries-preview/wiring.ts [hunk 6]
```diff
@@ -1,8 +1,64 @@
...
+ * @public
+ */
+export const fetchAndSaveQueryPreviewWire = wireDispatch('fetchAndSaveQueryPreview');
+
+/**
+ * Clears a query preview from queries preview module.
+ *
+ * @public
+ */
+export const clearQueryPreview = wireCommit('clearQueryPreview');
```
Comment (tiborux) [inline packages/x-components/src/x-modules/queries-preview/wiring.ts:41]: Add new empty line.

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    it('should request and store query preview results in the state', async () => {
+      const query = 'tshirt';
+      const request = getQueryPreviewRequest(query);
+
+      const promise = store.dispatch('fetchAndSaveQueryPreview', request);
```
Comment (tiborux) [inline packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:68]:
```suggestion
      await store.dispatch('fetchAndSaveQueryPreview', request);
```

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  localVue.use(Vuex);
+
+  const store: SafeStore<
+    QueriesPreviewState,
+    QueriesPreviewGetters,
+    QueriesPreviewMutations,
+    QueriesPreviewActions
+  > = new Store(queriesPreviewXStoreModule as any);
+  installNewXPlugin({ adapter, store }, localVue);
+
```
Comment (tiborux) [inline packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:40]: Use the util that you created, right now is not. being used

```
beforeEach(() => {
    resetQueriesPreviewStateWith(store);
  });
  ```

**Review by tiborux — CHANGES_REQUESTED** [review]
gj

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+  describe('fetchQueryPreview', () => {
+    it('should make a search request with a unique id', async () => {
+      const request = getQueryPreviewRequest('sandals');
```
Comment (tiborux) [inline packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:43]: This test and the next one can be combined in one.

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts (hunk 4 above)
Comment (mnavarroespinosa) [reply packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:41]: You're right! I actually added the util to reset it but forgot to use it in the tests 😅. I've added it!

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts (hunk 9 above)
Comment (mnavarroespinosa) [reply packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts:49]: Sure, I've combined them.

File: packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts (hunk 3 above)
Comment (mnavarroespinosa) [reply packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts]: Sure! I've change it, it did not make sense.

**Review by javieri-empathy — APPROVED** [review]

**Review by tiborux — APPROVED** [review]

**Pull request merged**